pipeline/coaddition.py

```diff
@@ -404,13 +422,18 @@ def run(self, images):
             upstreams=output.get_upstream_provenances(),
             process='coaddition',
         )
+        output.provenance_id = output.provenance.id
+
         output.is_coadd = True
         output.new_image = None
 
+        if aligned_images is not None:
+            output.aligned_images = aligned_images
+
         if self.pars.method == 'naive':
-            outim, outwt, outfl = self._coadd_naive(images)
+            outim, outwt, outfl = self._coadd_naive(output.aligned_images)
```

Comment:
If you leave the aligned_images parameter to run as None, how does output.aligned_images get filled?  It looks to me like it doesn't.

Reply:
The `output` image is generated from a list of upstream images, and has, in its provenance, an "alignment" parameter dictionary. Using those the output image can lazy-calculate aligned images. 
The ability of the pipeline to accept aligned images as an optional input is to skip this calculation and save runtime if you already have the aligned images (e.g., when you cache them in tests).